### DISCUSSION TITLE
Video Capture Configuration (Gitter repost)

### DISCUSSION THREAD

**Discussion by rbgtoffolo** [praxis-live/support#127] [General]
HI everyone!!! Sorry for a dummy question!!
I'm starting with PraxisLive (amazing stuff!!!), but I'm struggling with Video Configuration on Linux (Ubuntu 22.04).

The video streamimg working fine with my Gstreamer-1.0 config. In command line, the v4l2src working fine as well.
But I can figure out how to setup PraxisLive Video Options to get webcam streamimg.
When I try to create a video:capture I get a "Error creating component"

The device config in Video options is: v4l2src device=/dev/video0

Really Tnks!!!

> Neilcsmith gently reply:
> 
> I'm curious that it fails to create a video:capture component at all. What about video:player? Could you share the output of apt list --installed  | grep gst?
> 

apt list results:

```
gir1.2-gst-plugins-bad-1.0/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
gir1.2-gst-plugins-base-1.0/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado,automático]
gir1.2-gstreamer-1.0/jammy-updates,now 1.20.3-0ubuntu1 amd64 [instalado,automático]
gstreamer1.0-adapter-pulseeffects/jammy,now 4.8.4-1 amd64 [instalado,automático]
gstreamer1.0-alsa/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado]
gstreamer1.0-autogain-pulseeffects/jammy,now 4.8.4-1 amd64 [instalado,automático]
gstreamer1.0-clutter-3.0/jammy-updates,now 3.0.27-2ubuntu1 amd64 [instalado,automático]
gstreamer1.0-convolver-pulseeffects/jammy,now 4.8.4-1 amd64 [instalado,automático]
gstreamer1.0-crystalizer-pulseeffects/jammy,now 4.8.4-1 amd64 [instalado,automático]
gstreamer1.0-gl/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado,automático]
gstreamer1.0-gtk3/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
gstreamer1.0-libav/jammy-updates,now 1.20.3-0ubuntu1 amd64 [instalado,automático]
gstreamer1.0-opencv/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado]
gstreamer1.0-pipewire/jammy-updates,now 0.3.48-1ubuntu3 amd64 [instalado,automático]
gstreamer1.0-plugins-bad/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
gstreamer1.0-plugins-base-apps/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado]
gstreamer1.0-plugins-base/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado,automático]
gstreamer1.0-plugins-base/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 i386 [instalado,automático]
gstreamer1.0-plugins-good/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
gstreamer1.0-plugins-ugly/jammy,now 1.20.1-1 amd64 [instalado]
gstreamer1.0-pulseaudio/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado]
gstreamer1.0-tools/jammy-updates,now 1.20.3-0ubuntu1 amd64 [instalado,automático]
gstreamer1.0-vaapi/jammy-updates,now 1.20.1-1ubuntu1 amd64 [instalado,automático]
gstreamer1.0-x/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado,automático]
libclutter-gst-3.0-0/jammy-updates,now 3.0.27-2ubuntu1 amd64 [instalado,automático]
libgst7/jammy,now 3.2.5-1.3ubuntu1 amd64 [instalado,automático]
libgstreamer-gl1.0-0/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado,automático]
libgstreamer-opencv1.0-0/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
libgstreamer-plugins-bad1.0-0/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
libgstreamer-plugins-base1.0-0/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado,automático]
libgstreamer-plugins-base1.0-0/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 i386 [instalado,automático]
libgstreamer-plugins-base1.0-dev/jammy-updates,jammy-security,now 1.20.1-1ubuntu0.1 amd64 [instalado]
libgstreamer-plugins-good1.0-0/jammy-updates,jammy-security,now 1.20.3-0ubuntu1.1 amd64 [instalado,automático]
libgstreamer1.0-0/jammy-updates,now 1.20.3-0ubuntu1 amd64 [instalado,automático]
libgstreamer1.0-0/jammy-updates,now 1.20.3-0ubuntu1 i386 [instalado,automático]
libgstreamer1.0-dev/jammy-updates,now 1.20.3-0ubuntu1 amd64 [instalado]
libqt5multimediagsttools5/jammy,now 5.15.3-1 amd64 [instalado,automático]
```

Thnks Neil...

**Comment by neilcsmith-net** [comment]
Thanks for sharing that list.  Although nothing particular stands out.  I think you've got every package I have installed locally, plus a few extra.

Does a `video:player` component work OK?

Bear in mind that the default pipeline for `video:capture` is using `autovideosrc`.  And it tries to create this pipeline when creating the component - probably the wrong approach and needs fixing!  If something in the automatic source resolution fails in GStreamer this could be why you're not able to create the component.

What happens if you run 

```bash
gst-launch-1.0 autovideosrc ! videorate ! videoscale ! videoconvert ! appsink
```

You shouldn't see a window, but a pipeline should start and a clock show up in the CLI.

If that fails, try running with debug output -

```bash
GST_DEBUG=4 gst-launch-1.0 autovideosrc ! videorate ! videoscale ! videoconvert ! appsink
```

You could also try running PraxisLIVE from the CLI with the same setting 

```bash
GST_DEBUG=4 praxislive
```

You should see debug output from GStreamer in the Output window in the IDE when you create a `video:capture`.

**Reply by rbgtoffolo** [reply]
Hi Neil,

Hello Neil,

Yes, the video:player works perfectly.

Using your debugging tips for Gstreamer, I found an issue with the autovideosrc module conflicting with pipewire in the GStreamer 1.20.x series, which is still open, as we can see [here](https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/104)

So, I tested it on another machine with GStreamer 1.22, and everything worked correctly.

The solution, as I've seen with people who've had similar issues, is to compile the latest GStreamer from source and install it in an alternative location. This is because, as far as I recall, updating GStreamer on Ubuntu 22.04 can potentially break many things in the system.

Thank you very much for the tips and the brilliant work!

**Reply by neilcsmith-net** [reply]
Thanks for letting me know that worked.  While that might be a solution, I would also try -

```bash
 GST_PLUGIN_FEATURE_RANK=pipewiresrc:NONE praxislive
```

I've found Pipewire causing a bunch of issues with GStreamer code.

**Reply by rbgtoffolo** [reply]
Hi again!

Worked perfectly!

Closing discussion!!!

Tnks one more time!!!!
